File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py

def solution(elements):
    result = set()
    
    elementLen = len(elements)
    
    # [7, 9, 1, 1, 4, 7, 9, 1, 1, 4]
    elements = elements * 2
    
    # 길이가 i인 연속 부분 수열 개수 구하기
    for i in range(elementLen):
        for j in range(elementLen):
            result.add(sum(elements[j:j+i+1]))
    return len(result)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        Set<Integer> answer = new HashSet<Integer>();
        int length = elements.length;
        int[] lq = new int[length + length];
        
        
        for(int i = 0; i < lq.length; i++){
            lq[i] = elements[i % length];
        }

        for(int i = 1; i <= length; i++){
            for (int j = 0; j < length; j++){
                int sum = 0;
                for(int k = 0; k < i; k++){
                    sum += lq[k + j];
                }
                answer.add(sum);
            }
        }
        return answer.size();
    }
}